### ISSUE TITLE
genie_python: update 2.7 and dependencies

### ISSUE THREAD

**Issue by mattclarke** [ISISComputingGroup/IBEX#2548]
We should probably be using a more recent version of 2.7 and the various dependencies.

This should be relatively simple to do. It is mostly downloading stuff and editing the build batch file.

I think we would benefit from using parts of the WinPython package to create the core.

**Comment by John-Holt-Tessella** [comment]
This should includes pythongit which we have a had problem with.

**Comment by AdrianPotter** [comment]
#2624: There are issues with the updated Python Git with committing modified settings files. With the old Python Git it was failing with an error. I will leave resolving these issues to the associated ticket

**Comment by AdrianPotter** [comment]
#2386: I've updated to 1.1.1. There's a fix that we want in 1.2. We should apply the update when available but it shouldn't impede the remainder of the ticket

**Comment by Tom-Willemsen** [comment]
Doesn't build on windows 10. Adrian will look at using an installer instead of zip file

**Comment by Tom-Willemsen** [comment]
- Error importing matplotlib:
```
%matplotlib qt4 
set_instrument('TE:NDW1799:')

THIS IS NDW1799! (test machine)
There was a problem with loading init_ndw1799 so will use default.
Error was No module named init_ndw1799
PV prefix is TE:NDW1799:
  File "<input>", line 1
    %matplotlib qt4
    ^
SyntaxError: invalid syntax
```

- Exception while running script:
```
C:\Instrument\Apps\Python\lib\site-packages\CaChannel\CaChannel.py:422: RuntimeWarning: tp_compare didn't return -1 or -2 for exception
  if status != ca.ECA_NORMAL:
```

- Error on running g.end():
```
g.end()
** Ending Run 00000387 at 10:25:00 29/09/17 
ERROR: integer argument expected, got float
```

**Comment by AdrianPotter** [comment]
Install script needed to include Visual Studio C++ compiler for Python 2.7. We didn't need it before because we packaged Python differently. I didn't notice during testing because it was already installed on my machine

**Comment by Tom-Willemsen** [comment]
pydev console doesn't launch

**Comment by AdrianPotter** [comment]
Has something to do with our home-grown version of IPython. PyDev console doesn't run with IPython version 5.5 installed from PyDev as on the current branch, or even version 2.2 which we apparently use at the moment

**Comment by Tom-Willemsen** [comment]
Still unable to use `g.end()`

```
g.end()
** Ending Run 00000440 at 17:14:21 23/10/17 
ERROR: integer argument expected, got float
C:\Instrument\Apps\Python\lib\site-packages\CaChannel\CaChannel.py:422: RuntimeWarning: tp_compare didn't return -1 or -2 for exception
  if status != ca.ECA_NORMAL:
```

**Comment by AdrianPotter** [comment]
The previous issue was caused because the updated CaChannel API does not accept floats to enums. I have updated our own calls to genie_python so we're passing the correct values. We won't necessarily catch this in user scripts. Instrument scientists should be informed of the new restrictions. I've added it to the release notes.

I've added a ticket #2708 to look at creating an automated system test framework for genie_python

I've added a test to the manual system tests to run to verify some of the genie_python behaviour ahead of deployment.
